Sming/SmingCore/Network/Ssl/SslValidator.h

```diff
@@ -16,20 +16,40 @@
 #ifndef _SMING_CORE_SSLVALIDATOR_H_
 #define _SMING_CORE_SSLVALIDATOR_H_
 
-#ifdef ENABLE_SSL
-
 #include "ssl/ssl.h"
 #include "ssl/tls1.h"
 
 #include <functional>
+#include "WVector.h"
 
 typedef std::function<bool(SSL* ssl, void* data)> SslValidatorCallback;
 
 bool sslValidateCertificateSha1(SSL* ssl, void* data);
 
 bool sslValidatePublicKeySha256(SSL* ssl, void* data);
 
-#endif /* ENABLE_SSL */
+struct SSLValidator {
+	SslValidatorCallback callback;
```

Comment:
Could ditch callbacks and use `SslFingerprintType` with `switch`, can then include length check

Reply:
Fingerprints are just one small part of the things that you can check in a certificate and decide if it is valid or not. The callback is there to provide more flexibility

Reply:
Can you give me some examples? I'm just trying to understand this fully in case I'm over-simplifying.

Reply:
> Can you give me some examples? I'm just trying to understand this fully in case I'm over-simplifying.

Let me see if I can do this. 

A developer may want to have the fingerprint checks and in addition have a validator that checks some properties of the certificate. 
In the callback the developer will get access to the SSL certificate. Such a certificate contains multiple data objects (https://www.cryptologie.net/article/262/what-are-x509-certificates-rfc-asn1-der/) and for example there can be SSL validation check that makes sure that the length of the public key used in the certificate is greater than 1024 in order to reject weak keys.

Reply:
Thanks, that helps.